airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialReadUtil.java

```diff
@@ -1,18 +1,281 @@
+      LOGGER.info("Streams to be synced via ordered column : {}", initialLoadStreams.streamsForInitialLoad().size());
+      LOGGER.info("Streams: {}", prettyPrintConfiguredAirbyteStreamList(initialLoadStreams.streamsForInitialLoad()));
+      final MssqlInitialLoadStateManager initialLoadStateManager =
+          new MssqlInitialLoadGlobalStateManager(initialLoadStreams,
+              initPairToOrderedColumnInfoMap(database, initialLoadStreams, tableNameToTable, quoteString),
+              stateToBeUsed, catalog);
+      final MssqlDebeziumStateAttributes stateAttributes = MssqlDebeziumStateUtil.getStateAttributesFromDB(database);
+
+      final MssqlInitialLoadSourceOperations sourceOperations =
+          new MssqlInitialLoadSourceOperations();
+
+      final MssqlInitialLoadHandler initialLoadHandler = new MssqlInitialLoadHandler(sourceConfig, database,
+          sourceOperations, quoteString, initialLoadStateManager,
+          namespacePair -> Jsons.emptyObject(),
+          getTableSizeInfoForStreams(database, initialLoadStreams.streamsForInitialLoad(), quoteString));
+
+      initialLoadIterator.addAll(initialLoadHandler.getIncrementalIterators(
+          new ConfiguredAirbyteCatalog().withStreams(initialLoadStreams.streamsForInitialLoad()),
+          tableNameToTable,
+          emittedAt));
+    } else {
+      LOGGER.info("No streams will be synced via ordered column");
+    }
+
+    // Build the incremental CDC iterators.
+    final var targetPosition = MssqlCdcTargetPosition.getTargetPosition(database, sourceConfig.get(JdbcUtils.DATABASE_KEY).asText());
+    final AirbyteDebeziumHandler<Lsn> handler = new AirbyteDebeziumHandler<>(
+        sourceConfig,
+        targetPosition,
+        true,
+        firstRecordWaitTime,
+        subsequentRecordWaitTime,
+        OptionalInt.empty());
+
+    final Supplier<AutoCloseableIterator<AirbyteMessage>> incrementalIteratorsSupplier = () -> handler.getIncrementalIterators(catalog,
+        new MssqlCdcSavedInfoFetcher(stateToBeUsed),
+        new MssqlCdcStateHandler(stateManager),
+        mssqlCdcConnectorMetadataInjector,
+        getDebeziumProperties(database, catalog, false),
+        DebeziumConnectorType.RELATIONALDB,
+        emittedAt,
+        true); // TODO: check why add db name to state
+
+    // This starts processing the transaction logs as soon as initial sync is complete, this is a bit
+    // different
+    // from the current cdc syncs.
+    // We finish the current CDC once the initial snapshot is complete and the next sync starts
+    // processing the transaction logs
+    return Collections.singletonList(
+        AutoCloseableIterators.concatWithEagerClose(
+            Stream
+                .of(initialLoadIterator, Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorsSupplier, null)))
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList()),
```

Comment:
nit: `.toList()`

Reply:
The case to watch for here is in case the retuned list has to be modified for whatever reason (we had something similar in Postgres with vacuum).
.toList() returns an immutable list.

Reply:
Right, and it was a deliberate choice by the language maintainers to make that the default, because unless you're certain about the ownership of some state you're almost always better off making a modified copy rather than mutating it in-place.